lib/validator.js

```diff
@@ -506,6 +506,66 @@ Specberus.prototype.getDelivererIDsNote = function () {
     return ids;
 };
 
+Specberus.prototype.getChartersAndPolicy = async function () {
+    if (undefined !== this.charters) {
+        return {
+            charters: this.charters,
+            patentPolicy: this.patentPolicy,
+        };
+    }
+
+    const deliverers = await this.getDelivererIDs();
+    const docDate = this.getDocumentDate();
+    let groupsCharters = [];
+    const patentPolicy = [];
+    const currentCharters = [];
+
+    if (deliverers.length) {
+        w3capi.apiKey = process.env.W3C_API_KEY;
+
+        const delivererPromises = [];
+
+        const { TAG_ID } = util;
+        // Get charters data from W3C API
+        // deliverers.forEach is for joint publication.
+        deliverers.forEach((deliverer) => {
+            // Skip finding charter for TAG group.
+            if (deliverer === TAG_ID) return;
+            delivererPromises.push(
+                new Promise((resolve) => {
+                    w3capi
+                        .group(deliverer)
+                        .charters()
+                        .fetch({ embed: true }, (err, charters) => {
+                            resolve(charters);
+                        });
+                })
+            );
+        });
+        groupsCharters = await Promise.all(delivererPromises);
+        // Find the current charter, and curresponding patentPolicy.
+        // groups -> group is for joint publication.
+        groupsCharters.forEach((groupCharters) => {
+            groupCharters.forEach((groupCharter) => {
+                if (
+                    docDate >= new Date(groupCharter.start) &&
+                    docDate <= new Date(groupCharter.end)
+                ) {
+                    patentPolicy.push(groupCharter['patent-policy']);
+                    currentCharters.push(groupCharter.uri);
+                }
+            });
+        });
+    }
+
+    this.charters = currentCharters;
+    [this.patentPolicy] = patentPolicy;
```

Comment:
If there's a joint publication, we should check that all the deliverers are operating under the same patent policy.
With the current code, there will be no error if that's not the case and only the first PP will be returned.